sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,315 @@
+
+== Definitions
+
+
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+template<auto& f, class propertiesT = decltype(oneapi::experimental::properties{})>
+struct task_sequence{};
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...),
+          class propertiesT>
+class task_sequence<f, propertiesT> {
+
+public:
+  task_sequence(const task_sequence &) = delete;
+  task_sequence &operator=(const task_sequence &) = delete;
+  task_sequence(task_sequence &&) = delete;
+  task_sequence &operator=(task_sequence &&) = delete;
+
+  task_sequence();
+
+  task_sequence(const propertiesT &);
+
+  void async(ArgsT... args);
+
+  ReturnT get();
+  
+  template <typename propertyT>
+  static constexpr bool has_property();
+  
+  template <typename propertyT>
+  static constexpr auto get_property();
+
+  ~task_sequence();
+};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+
+----
+
+`task_sequence` is a class template, parameterized by an `auto` reference to a +Callable+ `f`. The +Callable+ `f` defines the asynchronous task to be associated with the `task_sequence`, and requiring an auto reference ensures that each `f` be statically resolvable at compile time.  Static resolvability by the compiler is desirable when compiling for spatial architectures as it can enable the generation of more efficient hardware. Furthermore, the partial specialization `typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...)` ensures that the return type `ReturnT` of `f` and the argument types `typename ... ArgsT` of `f` are resolvable and fixed for each definition of `task_sequence`. 
+
+The `task_sequence` interface consists of two methods, `async` and `get`. The `async` method asynchronously invokes `f`, and stores the return value into a FIFO queue upon completion of `f`.  The `async` function call is non-blocking in that it returns before the asynchronous `f` invocation completes executing, and potentially before `f` even begins executing (return from the `async` provides no implicit information on the execution status of `f`). The `get` method retrieves the oldest result from this logical FIFO queue, and blocks (waits) until a result is available if none are available immediately upon the call to `get`. Both methods may only be invoked on the device on which a `task_sequence` object has been instantiated. Calling `async` or `get` on a different device results in undefined behavior.
+
+[frame="topbot",options="header"]
+|===
+|Functions |Description
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+task_sequence(const propertyListT &);
+
+task_sequence();
+----
+|
+Constructor for `task_sequence`. 
+
+Accepts properties `balanced`, `invocation_capacity`, and `response_capacity` defined below
```

Comment:
We do not need to reference every property that this class uses here in the description of the ctor.

Reply:
Removed.